### PR TITLE
Finished the fire and hire, now making pull request

### PR DESCRIPTION
**Pull request by IsisJ**

### CODE REVIEW

File: src/HireAndFire/HireAndFire.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+public class HireAndFire implements ActionListener {
+
+	private JPanel panel = new JPanel();
+	private JButton hireButton = new JButton();
+	private JButton fireButton = new JButton();
+	private JFrame frame = new JFrame();
+	private JTextField hireTextField = new JTextField(10);
+	private JTextField fireTextField = new JTextField(10);
+	private Stack<String> employees = new Stack<String>();
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:21]: nice use of private!

Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:21]: Used descriptive variable names 😸

File: src/HireAndFire/HireAndFire.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		panel.add(fireButton);
+		
+		hireButton.addActionListener(this);
+		hireButton.setText("HIRE");
+		fireButton.addActionListener(this);
+		fireButton.setText("FIRE");
+		frame.setVisible(true);
+		frame.setSize(300,100);
+		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	}
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:37]: very clear organization!

File: src/HireAndFire/HireAndFire.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	private JPanel panel = new JPanel();
+	private JButton hireButton = new JButton();
+	private JButton fireButton = new JButton();
+	private JFrame frame = new JFrame();
+	private JTextField hireTextField = new JTextField(10);
+	private JTextField fireTextField = new JTextField(10);
+	private Stack<String> employees = new Stack<String>();
+	
+	private void createUI() {
+
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java]: extra space

File: src/HireAndFire/HireAndFire.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+		hireButton.setText("HIRE");
+		fireButton.addActionListener(this);
+		fireButton.setText("FIRE");
+		frame.setVisible(true);
+		frame.setSize(300,100);
+		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	}
+	
+	
+	@Override
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:39]: can delete @ override

File: src/HireAndFire/HireAndFire.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonPressed = (JButton)e.getSource();
+		if(buttonPressed == hireButton) {
+
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java]: can delete space

File: src/HireAndFire/HireAndFire.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	
+	
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonPressed = (JButton)e.getSource();
+		if(buttonPressed == hireButton) {
+			
+			String theHiree = hireTextField.getText();
+			employees.push(theHiree);
+			hireTextField.setText(null);
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:45]: i like how you've cleared the text field!

File: src/HireAndFire/HireAndFire.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		}
+		else if(buttonPressed == fireButton){
+
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java]: can delete space

File: src/HireAndFire/HireAndFire.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+		else if(buttonPressed == fireButton){
+			
+			String theFireList = "";
+			int howManyToFire = Integer.parseInt(fireTextField.getText());
+			if(howManyToFire > employees.size()) {
+				JOptionPane.showMessageDialog(frame, "You only have "+employees.size()+" employees.");
+			}
+			else {
+				for(int i = 0; i<howManyToFire; i++) {
+					theFireList = theFireList +"\n"+employees.pop();
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:55]: would putting \n before employees.pop make an extra blank return before listing off the employees that have been fired?

File: src/HireAndFire/HireAndFire.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+				}
+				JOptionPane.showMessageDialog(frame, theFireList);
+			}
+			fireTextField.setText(null);
+		}
+
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java]: remove extra spaces here

File: src/HireAndFire/HireAndFire.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+				JOptionPane.showMessageDialog(frame, theFireList);
+			}
+			fireTextField.setText(null);
+		}
+		
+	}
+	
+	public static void main(String[] args) {
+	
+		HireAndFire manager = new HireAndFire();
```
Comment (epuzio) [inline src/HireAndFire/HireAndFire.java:64]: i like how the class in charge of hiring and firing is called manager 🍐 🍑 🍎

File: src/HireAndFire/HireAndFire.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+		if(buttonPressed == hireButton) {
+			
+			String theHiree = hireTextField.getText();
+			employees.push(theHiree);
+			hireTextField.setText(null);
+		}
+		else if(buttonPressed == fireButton){
+			
+			String theFireList = "";
+			int howManyToFire = Integer.parseInt(fireTextField.getText());
```
Comment (DCamaron1) [inline src/HireAndFire/HireAndFire.java:49]: great use of names

File: src/HireAndFire/HireAndFire.java (hunk 9 above)
Comment (DCamaron1) [inline src/HireAndFire/HireAndFire.java]: please remove the extra spaces

File: src/HireAndFire/HireAndFire.java (hunk 4 above)
Comment (DCamaron1) [inline src/HireAndFire/HireAndFire.java:39]: Get rid of the Override and spaces 🍔

File: src/HireAndFire/HireAndFire.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+			if(howManyToFire > employees.size()) {
```
Comment (DCamaron1) [inline src/HireAndFire/HireAndFire.java:50]: Great job on this

**Pull request merged**